graspy/__init__.py

```diff
@@ -1,3 +1,6 @@
+# Copyright (c) Microsoft Corporation and contributors.
+# Licensed under the MIT License.
+
 import warnings
```

Comment:
We specifically do not want to suppress these warnings in our init.  We can expose a utility that will do it for us if we want, but changing the global warning system just by using our library is not something we want to do.  Changing the global warnings or logging is not something we want to do (even if other projects do it).  What these libraries don't understand is they're not responsible for configuring a logger or the warning system, the final consumer/user is.

Reply:
Just something I noticed when I was adding this header :)

Reply:
Yea this makes sense. with larger packages like sklearn and numpy, theres often deprecation warnings which can be annoying but it is waht it is.

Reply:
i am in favor of some function-specific warning suppression, though, and can explain why. but agree that we don't need to do anything at all globally on init see #464

Reply:
an example im talking about is if we run something that throws a `ConvergenceWarning` but if it doesn't converge we re-run it internally anyway, and then that converges. The user doesn't need to or want to (IMO) know that some random initialization we ran didn't converge, so i know we suppress a convergence warning for a specific function somewhere. and that use case makes sense to me

Reply:
yeah, I agree, especially if we can grab the warning state on ConvergenceWarning first, set it as a temp early in the function, set it to suppress while we run our stuff, then in a finally block in the function set it back to the initial state - cleaning up after ourselves, if you will.